plugins/auth-react/src/components/AppAuthProvider/AppAuthProvider.tsx

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ReactNode } from 'react';
+import { CookieAuthRefreshProvider } from '@backstage/plugin-auth-react';
+import { isProtectedApp } from './isProtectedApp';
+
+/**
+ * @public
+ * A provider that will protect the app when running in protected experimental mode.
+ */
+export function AppAuthProvider(props: { children: ReactNode }): JSX.Element {
```

Comment:
Hmm, this doesn't need to live here right? We can move all of this over to `core-app-api`, including `isProtectedApp`. And then use the regular relative import to share it with `frontend-app-api`.

Reply:
Done